SDBlockDevice.cpp

```diff
@@ -147,9 +147,17 @@
 #include "mbed_debug.h"
 #include <errno.h>
 
-/* Required version: 5.5.4 and above */
-#if (MBED_VERSION < MBED_ENCODE_VERSION(5,5,4))
-#error "Incompatible mbed-os version detected! Required 5.5.4 and above"
+#ifndef MBED_VERSION_CHECK
+/* MBED_VERSION_CHECK macro is to used by applications / library to verify compatibility with mbed-os
+ * @return 1 - OS version compatible i.e. mbed-os version is same as macro argument or higher
+ * @return 0 - OS version lower than macro arguments
+ * To be moved to mbed-os code */
+#define MBED_VERSION_CHECK(major, minor, patch) \
+        (MBED_ENCODE_VERSION(MBED_MAJOR_VERSION, MBED_MINOR_VERSION, MBED_PATCH_VERSION) >= MBED_ENCODE_VERSION(major, minor, patch) ? 1 : 0)
+#endif
+
+#if !MBED_VERSION_CHECK(5,5,4)
```

Comment:
If this was in the OS as MBED_COMPATIBILITY_CHECK and didnt have to use ! it'd be great!